scripts/download-realm.js

```diff
@@ -97,38 +81,31 @@ function download(serverFolder, archive, destination) {
         const agentOpts = require('url').parse(proxyUrl);
         agent = new HttpsProxyAgent(agentOpts);
     }
-    return fetch(url, { agent }).then((response) => {
+    return fetch(url, {agent}).then(response => {
         if (response.status !== 200) {
             throw new Error(`Error downloading ${url} - received status ${response.status} ${response.statusText}`);
         }
 
         const lastModified = new Date(response.headers.get('Last-Modified'));
-        return fs.exists(destination)
-                 .then(exists => {
-                     if (!exists) {
-                         return saveFile();
-                     } else {
-                         return fs.stat(destination)
-                                  .then(stat => {
-                                      if (stat.mtime.getTime() !== lastModified.getTime()) {
-                                          return saveFile();
-                                      }
-                         })
-                     }
-        });
-
-        function saveFile() {
-            if (process.stdout.isTTY) {
-                printProgress(response.body, parseInt(response.headers.get('Content-Length')), archive);
-            } else {
-                console.log(`Downloading ${archive}`);
+        const contentLength = response.headers.get('Content-Length');
+        return fs.stat(destination).then(stat => {
+            if (stat.mtime.getTime() <= lastModified.getTime() || stat.size !== +contentLength) {
+                return saveFile(response);
             }
-            return new Promise((resolve) => {
-                const file = fs.createWriteStream(destination);
-                response.body.pipe(file).once('finish', () => file.close(resolve));
-            }).then(() => fs.utimes(destination, lastModified, lastModified));
-        }
+        }).catch(() => saveFile(response));
```

Comment:
Why are we saving the file in case of an error?

Reply:
This will hide the error, is that on purpose? I would `console.error` it at the least.

Reply:
fs.stat() throws if the file does not exist or if the user doesn't have permission to access it. The former case isn't an error and we just want to download the file, and in the latter case saveFile will throw an error anyway.